plugins/modules/iam_role_info.py

```diff
@@ -283,7 +216,28 @@ def main():
         collection_name="amazon.aws",
     )
 
-    module.exit_json(changed=False, iam_roles=describe_iam_roles(module, client))
+    # Once the deprecation is over we can merge this into a single call to validate_iam_identifiers
+    if name:
+        validation_error = validate_iam_identifiers("role", name=name)
+        if validation_error:
+            module.fail_json(msg=validation_error)
+    if path_prefix:
+        validation_error = validate_iam_identifiers("role", path=path_prefix)
+        if validation_error:
+            _prefix = "/" if not path_prefix.startswith("/") else ""
+            _suffix = "/" if not path_prefix.endswith("/") else ""
+            path_prefix = "{_prefix}{path_prefix}{_suffix}"
```

Comment:
Did you mean to use an f-string here? If you do it like this you'll get a `ValidationError`, because `path_prefix` starts with a litteral `{`:

> The specified value for pathPrefix is invalid. It must begin with the / character and contain only alphanumeric characters and/or / characters.

Reply:
Good catch.  Unfortunately our integration tests didn't check the use of a "bad" path so this one slipped through.  One of the reasons we want to move towards shared code for some of these pieces is that the unit tests for the logic can be much more thorough.

Reply:
@vonschultz should be fixed by #2066

7.6.0 is currently slated for the 7th of May, and we should be able to get the fix merged before that.

Work around is simply to pass the "current" prefix including a `/` at the start and end of the path.